libs/cardano-ledger-core/cddl/Cardano/Ledger/Huddle.hs

```diff
@@ -1,29 +1,55 @@
 {-# LANGUAGE AllowAmbiguousTypes #-}
 {-# LANGUAGE DataKinds #-}
+{-# LANGUAGE GADTs #-}
 {-# LANGUAGE KindSignatures #-}
 {-# LANGUAGE MultiParamTypeClasses #-}
-{-# LANGUAGE OverloadedStrings #-}
 {-# LANGUAGE ScopedTypeVariables #-}
 {-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE UndecidableInstances #-}
 {-# LANGUAGE UndecidableSuperClasses #-}
 
 module Cardano.Ledger.Huddle (
+  Type0 (..),
   HuddleRule (..),
   HuddleGroup (..),
-  HuddleGRule (..),
+  huddleRule,
+  huddleGroup,
   Era,
 ) where
 
 import Cardano.Ledger.Core (Era)
-import Codec.CBOR.Cuddle.Huddle
+import Codec.CBOR.Cuddle.Huddle (Group, IsType0, Named, Rule, comment, (=:=), (=:~))
 import Data.Proxy (Proxy (..))
-import GHC.TypeLits (KnownSymbol, Symbol)
+import qualified Data.Text as T
+import GHC.TypeLits (KnownSymbol, Symbol, symbolVal)
+
+data Type0 where
+  Type0 :: IsType0 t => t -> Type0
```

Comment:
I think this is fine as a placeholder, but we don't really need it once we switch to `cuddle-1.1.2.0`, then we can switch to just using `toType0` and use the `Type0` newtype exported from Huddle.